src/test/java/org/elasticsearch/index/query/MatchAllQueryBuilderTest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.search.ConstantScoreQuery;
+import org.apache.lucene.search.MatchAllDocsQuery;
+import org.apache.lucene.search.Query;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.*;
+
+public class MatchAllQueryBuilderTest extends BaseQueryTestCase<MatchAllQueryBuilder> {
+
+    @Override
+    public void assertLuceneQuery(MatchAllQueryBuilder queryBuilder) throws IOException {
+        Query query = queryBuilder.toQuery(context);
+        if (queryBuilder.boost() != 1.0f) {
+            assertThat(query, instanceOf(MatchAllDocsQuery.class));
+        } else {
+            assertThat(query, instanceOf(ConstantScoreQuery.class));
+        }
+        assertThat(query.getBoost(), is(queryBuilder.boost()));
+    }
+
+    @Override
+    public MatchAllQueryBuilder createEmptyQueryBuilder() {
```

Comment:
Actually I think I think we should have at least the `createTestQueryBuilder()` method public because my idea was to reuse these methods in other tests where we need randomized nested queries. Will reduce the visibility for the other ones if that's okay with you.